homework/index.js

```diff
@@ -30,15 +14,119 @@
+          class: 'contributor',
+          target: '_blank',
+        });
+        createAndAppend('div', li, {
+          text: contributor.contributions,
+          class: 'contribution-count',
+        });
       }
+    } else {
+      createAndAppend('p', contributorsContainer, {
+        text: 'No Contributors',
+        class: 'no-contributor',
+      });
+    }
+  }
+
+  function showRepositoryDetails(data) {
+    const repositoryContainer = document.getElementById('repository');
+    removeChildElements(repositoryContainer);
+    const selected = document.getElementById('repositorySelect').selectedIndex;
+    const repository = data[selected];
+    const h2 = createAndAppend('h2', repositoryContainer, { id: 'repositoryName' });
+    createAndAppend('a', h2, {
+      text: repository.name,
+      href: repository.html_url,
+      target: '_blank',
     });
+    const ul = createAndAppend('ul', repositoryContainer, { id: 'repositoryInfo' });
+    [
+      'Description',
+      repository.description,
+      'Forks',
+      repository.forks_count,
+      'Stargazers',
+      repository.stargazers_count,
+      'Watchers',
+      repository.watchers_count,
+      'Updated',
+      repository.updated_at,
+    ].forEach(value => createAndAppend('li', ul, { text: value }));
+
+    const contributorsContainer = document.getElementById('contributors');
+    fetch(repository.contributors_url)
+      .then(response => response.json())
+      .then(result => showContributors(result))
+      .catch(error => displayError(error, contributorsContainer));
+  }
+
+  function createHeader(data) {
+    const nav = createAndAppend('nav', root, { id: 'navigation', class: 'alignCenter' });
+    createAndAppend('img', nav, {
+      src: 'hyf.png',
+      alt: 'Hack Your Future Logo',
+      class: 'logo',
+    });
+    createAndAppend('label', nav, {
+      text: 'Hack Your Future Repositories:',
+      for: 'repositorySelect',
+    });
+
+    const select = createAndAppend('select', nav, { id: 'repositorySelect' });
+    data.sort((repo1, repo2) => repo1.name.localeCompare(repo2.name, 'en', { sensivity: 'base' }));
+    data.forEach((repository, index) => {
+      createAndAppend('option', select, {
+        text: repository.name,
+        value: repository.name,
+        id: index,
+      });
+    });
+
+    const main = createAndAppend('main', root, { id: 'main' });
+    createAndAppend('div', main, { id: 'repository', class: 'boxes' });
+    createAndAppend('div', main, { id: 'contributors' });
+    showRepositoryDetails(data);
+    select.addEventListener('change', () => showRepositoryDetails(data));
+  }
+
+  function main(url) {
+    const root = document.getElementById('root');
+
+    fetch(url)
+      .then(response => {
+        if (response.ok) {
+          return response.json();
+        }
+        throw new Error(`Network Error: ${response.url} (${response.statusText})`);
+      })
+      .then(result => {
+        createHeader(result);
```

Comment:
Small thing but you can also replace this then block like `.then(createHeader).catch(.......)` and it would work just the same 🙂